lib/puppet/provider/exec/pwsh.rb

```diff
@@ -17,19 +17,45 @@
   def run(command, check = false)
     @pwsh ||= get_pwsh_command
     self.fail 'pwsh could not be found' if @pwsh.nil?
-    write_script(command) do |native_path|
-      # Ideally, we could keep a handle open on the temp file in this
-      # process (to prevent TOCTOU attacks), and execute powershell
-      # with -File <path>. But powershell complains that it can't open
-      # the file for exclusive access. If we close the handle, then an
-      # attacker could modify the file before we invoke powershell. So
-      # we redirect powershell's stdin to read from the file. Current
-      # versions of Windows use per-user temp directories with strong
-      # permissions, but I'd rather not make (poor) assumptions.
-      if Puppet::Util::Platform.windows?
-        return super("cmd.exe /c \"\"#{native_path(@pwsh)}\" #{legacy_args} -Command - < \"#{native_path}\"\"", check)
-      else
-        return super("/bin/sh -c \"#{native_path(@pwsh)} #{legacy_args} -Command - < #{native_path}\"", check)
+    if PuppetX::PowerShell::PowerShellManager.supported_on_pwsh?
```

Comment:
I assume this code is straight copied from the `powershell` provider?

If so, `pwsh` and `powershell` are probably pretty close to one another and perhaps a refactor is in order to remove the duplication.

Reply:
Yeah potentially could make a shared function for this.

Reply:
Done!